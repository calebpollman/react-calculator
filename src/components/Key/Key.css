.key-container {
  align-items: center;
  display: flex;
  height: 25%;
  justify-content: center;
  transition: background-color 0.3s linear;
}

.key-container:hover {
  cursor: pointer;
}

.operator-key {
  background-color: var(--action-key-color);
  width: 100%;
}

.operator-key:hover {
  background-color: var(--action-key-color-hover);
}

.number-key {
  background-color: var(--number-key-color);
  width: calc(100% / 3);
}

.number-key:hover {
  background-color: var(--number-key-color-hover);
}

.submit-key {
  background-color: var(--submit-key-color);
  height: 100%;
  width: 100%;
}

.submit-key:hover {
  background-color: var(--submit-key-color-hover);
}

.key-value {
  color: var(--key-text-color);
  font-family: var(--main-font);
  font-size: var(--key-text-size);
  font-weight: var(--key-text-weight);
}
