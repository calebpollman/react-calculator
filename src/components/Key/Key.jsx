import React from 'react';
import PropTypes from 'prop-types';
import './Key.css';

const Key = ({ keyAction, keyType, keyValue }) => (
  <div
    className={`key-container ${keyType}`}
    onClick={() => keyAction(keyValue)}
  >
    <p className="key-value">
      {keyValue}
    </p>
  </div>
);

Key.propTypes = {
  keyAction: PropTypes.func.isRequired,
  keyType: PropTypes.string.isRequired,
  keyValue: PropTypes.string.isRequired,
}

export default Key;
