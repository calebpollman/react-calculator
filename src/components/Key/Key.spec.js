import { shallow } from 'enzyme';
import Key from './Key';

const MOCK_HANDLE_KEY_PRESS = jest.fn();
const MOCK_KEY_ACTION = jest.fn();
const MOCK_KEY_TYPE = 'number-key';
const MOCK_KEY_VALUE = '7';

describe('Key', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Key
        handleKeyPress={MOCK_HANDLE_KEY_PRESS}
        keyAction={MOCK_KEY_ACTION}
        keyType={MOCK_KEY_TYPE}
        keyValue={MOCK_KEY_VALUE}
      />
    );
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('should render the value of keyValue', () => {
    wrapper.setProps({ keyValue: 'test' });
    expect(wrapper.text()).toEqual('test');
  });
});
