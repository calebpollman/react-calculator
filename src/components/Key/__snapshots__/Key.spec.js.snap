// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Key should render correctly 1`] = `
<div
  className="key-container number-key"
  onClick={[Function]}
  onKeyPress={[Function]}
>
  <p
    className="key-value"
  >
    7
  </p>
</div>
`;
